.azure/pipelines/jobs/default-build.yml

```diff
@@ -287,7 +287,11 @@ jobs:
         condition: and(or(succeeded(), eq('${{ artifact.publishOnError }}', 'true')), or(eq(variables['system.pullrequest.isfork'], false), eq('${{ artifact.includeForks }}', 'true')))
         continueOnError: true
         inputs:
-          pathtoPublish: $(Build.SourcesDirectory)/${{ artifact.path }}
+          # Assume runtime variable values are absolute paths already.
+          ${{ if startsWith(artifact.path, '$(') }}:
+            pathToPublish: ${{ artifact.path }}
+          ${{ if not(startsWith(artifact.path, '$(')) }}:
+            pathToPublish: $(Build.SourcesDirectory)/${{ artifact.path }}
```

Comment:
The key word here is "runtime" i.e. the value of `artifact.path` for the Windows_arm_csharp artifact is `$(Build.ArtifactStagingDirectory)`, not e.g. `D:\a\_work\1\a`.